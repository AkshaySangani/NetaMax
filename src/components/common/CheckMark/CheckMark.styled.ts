import styled, { keyframes } from "styled-components";

const stroke = keyframes`
100% {
  stroke-dashoffset: 0;
}
`;

const scale = keyframes`
0%,
100% {
  transform: none;
}
50% {
  transform: scale3d(1.1, 1.1, 1);
}
`;

const fill = keyframes`
100% {
  box-shadow: inset 0 0 0 100vh #21C57C;
}
`;

export const Svg = styled.svg`
  display: block;
  margin-left: auto;
  margin-right: auto;
  border-radius: 50%;
  stroke: #fff;
  stroke-width: 5;
  stroke-miterlimit: 10;
  animation: ${fill} 0.4s ease-in-out 0.4s forwards, ${scale} 0.3s ease-in-out 0.9s both;
`;

export const Circle = styled.circle`
  stroke-dasharray: 166;
  stroke-dashoffset: 166;
  stroke-width: 5;
  stroke-miterlimit: 10;
  stroke: #fff;
  fill: none;
  animation: ${stroke} 0.6s cubic-bezier(0.65, 0, 0.45, 1) forwards;
`;

export const Path = styled.path`
  transform-origin: 50% 50%;
  stroke-dasharray: 48;
  stroke-dashoffset: 48;
  animation: ${stroke} 0.3s cubic-bezier(0.65, 0, 0.45, 1) 0.8s forwards;
`;
