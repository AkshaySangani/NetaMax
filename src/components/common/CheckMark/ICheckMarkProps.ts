export interface ICheckMarkProps {
  /**
   * The size of the checkMark.
   * @type {string}
   */
  size: "xSmall" | "small" | "medium" | "large" | "xLarge" | "xxLarge";

  /**
   * The color of the checkMark.
   * @type {string}
   */
  color: string;
}
