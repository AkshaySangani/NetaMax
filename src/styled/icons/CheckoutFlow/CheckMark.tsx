import { ReactElement } from "react";

/**
 * The icon for the checkout checkmark.
 * @returns {ReactElement} The icon.
 */
export default function CheckoutCheckmark(): ReactElement {
  return (
    <svg width="40" height="36" viewBox="0 0 40 36" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M38.2533 0.602678C36.4483 -0.369247 34.435 1.50518 33.2548 2.61595C30.5474 5.25403 28.2565 8.30865 25.6877 11.0856C22.8414 14.1402 20.2033 17.1948 17.2875 20.1801C15.6213 21.8463 13.8163 23.6513 12.7056 25.734C10.2063 23.3041 8.05422 20.666 5.27729 18.514C3.26402 16.9867 -0.0682945 15.8759 0.0011287 19.5553C0.139975 24.3456 4.37479 29.483 7.49883 32.7457C8.81787 34.1342 10.5535 35.5921 12.5667 35.6615C14.9965 35.8004 17.4958 32.8846 18.9537 31.2878C21.5224 28.5109 23.6051 25.3868 25.9654 22.5405C29.02 18.7917 32.1441 15.1121 35.1293 11.2938C37.0037 8.93346 42.9047 3.1018 38.2533 0.602678ZM3.05564 19.2776C2.98621 19.2776 2.91679 19.2776 2.77794 19.3469C2.50025 19.2776 2.29198 19.2081 2.01429 19.0692C2.22256 18.9304 2.56967 18.9998 3.05564 19.2776Z"
        fill="#353535"
      />
    </svg>
  );
}
