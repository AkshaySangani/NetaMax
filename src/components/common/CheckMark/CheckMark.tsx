import { ReactElement } from "react";

import { Circle, Path, Svg } from "./CheckMark.styled";
import { ICheckMarkProps } from "./ICheckMarkProps";

const PREDEFINED_SIZE_MAP = {
  xSmall: "16px",
  small: "24px",
  medium: "40px",
  large: "52px",
  xLarge: "72px",
  xxLarge: "96px",
};

/**
 *  The CheckMark component.
 * @param {ICheckMarkProps} props The props of the component.
 * @returns {ReactElement} The CheckMark component.
 */
export const Checkmark = (props: ICheckMarkProps): ReactElement => {
  const { size, color } = props;
  const computedSize = PREDEFINED_SIZE_MAP[size];
  const style: Record<string, string> = { width: computedSize, height: computedSize };
  if (color) {
    style["color"] = color;
  }

  return (
    <Svg className="checkmark" xmlns="http://www.w3.org/2000/svg" style={style} viewBox="0 0 52 52">
      <Circle className="checkmark__circle" cx="26" cy="26" r="25" fill="none" />
      <Path className="checkmark__check" fill="none" d="M14.1 27.2l7.1 7.2 16.7-16.8" />
    </Svg>
  );
};
