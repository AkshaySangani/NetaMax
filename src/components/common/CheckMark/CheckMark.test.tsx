import { render, cleanup } from "@testing-library/react";
import { Checkmark } from "./CheckMark";
import { ICheckMarkProps } from "./ICheckMarkProps";

describe("Checkmark Component", () => {
  let mock: ICheckMarkProps;
  beforeEach(() => {
    mock = {
      size: "xxLarge",
      color: "#eaf7ee",
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("Checkmark should be render correctly", () => {
    const { container } = render(<Checkmark {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("Custom size", () => {
    mock = {
      size: "small",
      color: "",
    };
    const { container } = render(<Checkmark {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
