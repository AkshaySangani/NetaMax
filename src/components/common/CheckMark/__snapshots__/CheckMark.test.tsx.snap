// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkmark Component Checkmark should be render correctly 1`] = `
<div>
  <svg
    class="sc-bdvvtL gAbBrJ checkmark"
    style="width: 96px; height: 96px; color: rgb(234, 247, 238);"
    viewBox="0 0 52 52"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      class="sc-gsDKAQ gXwmvj checkmark__circle"
      cx="26"
      cy="26"
      fill="none"
      r="25"
    />
    <path
      class="sc-dkPtRN dmPwJp checkmark__check"
      d="M14.1 27.2l7.1 7.2 16.7-16.8"
      fill="none"
    />
  </svg>
</div>
`;

exports[`Checkmark Component Custom size 1`] = `
<div>
  <svg
    class="sc-bdvvtL gAbBrJ checkmark"
    style="width: 24px; height: 24px;"
    viewBox="0 0 52 52"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      class="sc-gsDKAQ gXwmvj checkmark__circle"
      cx="26"
      cy="26"
      fill="none"
      r="25"
    />
    <path
      class="sc-dkPtRN dmPwJp checkmark__check"
      d="M14.1 27.2l7.1 7.2 16.7-16.8"
      fill="none"
    />
  </svg>
</div>
`;
